packages/backend/src/open_payments/payment/outgoing/service.ts

```diff
@@ -310,12 +308,12 @@ async function validateGrant(
       })
     ) {
       if (grantPayment.failed) {
-        const totalSent = await deps.accountingService.getTotalSent(
-          grantPayment.id
+        const totalSent = validateSentAmount(
+          deps,
+          payment,
+          await deps.accountingService.getTotalSent(grantPayment.id)
         )
-        if (totalSent === undefined) {
-          throw new Error()
```

Comment:
we have a few of these `throw new Error()` in the code, so trying to get rid of them step-by-step